tf/tfprocess.py

```diff
@@ -1229,69 +1429,49 @@ def construct_net(self, inputs):
                                           bias_regularizer=self.l2reg,
                                           name='policy/dense')(h_conv_pol_flat)
         elif self.POLICY_HEAD == pb.NetworkFormat.POLICY_ATTENTION:
-            # transpose and reshape
-            tokens = tf.transpose(flow, perm=[0, 2, 3, 1])
-            tokens = tf.reshape(tokens, [-1, 64, self.RESIDUAL_FILTERS])
-
+            if self.encoder_layers == 0:
+                attn_wts = []
+            if self.RESIDUAL_BLOCKS > 0:
+                # transpose and reshape
+                tokens = tf.transpose(flow, perm=[0, 2, 3, 1])
+                tokens = tf.reshape(tokens, [-1, 64, self.RESIDUAL_FILTERS])
+            else:
+                tokens = flow
             # SQUARE EMBEDDING: found to increase attention head performance
             tokens = tf.keras.layers.Dense(self.pol_embedding_size, kernel_initializer='glorot_normal',
                                            kernel_regularizer=self.l2reg, activation='selu',
                                            name='policy/embedding')(tokens)
-
-            # ENCODER LAYERS: intermediate layers of self-attention with residual connections
-            attn_wts = []
-            for i in range(self.pol_encoder_layers):
-                tokens, attn_wts_l = self.encoder_layer(tokens, self.pol_embedding_size, self.pol_encoder_d_model,
-                                                        self.pol_encoder_heads, self.pol_encoder_dff,
-                                                        name='policy/enc_layer_{}'.format(i + 1), training=True
-                                                        )
-                attn_wts.append(attn_wts_l)
+            if self.RESIDUAL_BLOCKS > 0:
```

Comment:
This implies we need an assert somewhere that we don't actually support the combination of encoders and policy encoders?
We might want an assert that residual blocks is 0 if encoders is non-zero and vice versa as well? (If I didn't miss it)

Reply:
line 149 should prevent a combination of body encoder and residual blocks
```
assert (self.RESIDUAL_BLOCKS > 0) != (self.encoder_layers > 0), \
                "Nets with both encoder layers and residual blocks are not supported"
```